### DISCUSSION TITLE
Any way to return JSON specific to the request?

### DISCUSSION THREAD

**Discussion by duyemura** [hotmeteor/receiver#17] [Receivers]
I might be missing it, but it seems like Receiver is best suited for handling webhooks asynchronously.  Is there any way to return any custom response back to the caller?  Everything I do leaves me with a blank json object?

I think it's ok if so, I can handle things asyc (even better really) but in some cases, I might want to handle sync and reply back with a payload of the action that took place.

**Comment by hotmeteor** [comment]
Ya, you could do something like this in your controller:

```php
return Receiver::driver($provider)
            ->receive($request);

return response()->json(... whatever you want here ...);
```

Basically it just ignores whatever happens and responds.

**Reply by duyemura** [reply]
I want it to respond back with the actual output/work of the webhook.  Whenever I run it - it seems to run the hook action async and thus every response is just {}

I think I'm likely doing something wrong.

**Reply by poms32** [reply]
Hi, did you manage to do it ? I'm facing the same issue !
